### PR TITLE
Fix open console command

### PR DESCRIPTION
**Pull request by sudhirverma**
Fix: #1247

### CODE REVIEW

File: src/k8s/console.ts [hunk 1]
```diff
@@ -27,8 +27,8 @@ export class Console extends OpenShiftItem {
             const consoleUrl = `${spec.port.targetPort}://${spec.host}`;
             return consoleUrl;
         } else {
-            const consoleUrl = await Console.odo.execute(Command.showServerUrl());
-            return consoleUrl.stdout;
+            const getUrlObj = await Console.odo.execute(Command.showConsoleUrl());
+            return JSON.parse(getUrlObj.stdout).data.consoleURL;
```
Comment (mohitsuman) [inline src/k8s/console.ts]: This change is specific to 3.x cluster and not 4.x cluster. In the issue you have mentioned that the 4.x cluster does not work properly.

Comment (sudhirverma) [reply src/k8s/console.ts]: okay @mohitsuman I will check.

Comment (sudhirverma) [reply src/k8s/console.ts]: old code works only when we login with kubeadmin but it doesn't work when we login with `oc login -u developer -p developer https://api.crc.testing:6443` for 4.x cluster.

now it works for both kubeadmin and for developer.

**Comment by dgolovin** [comment]
Will test on 3.x and 4.x clusters, before pushing to the master.

File: src/k8s/console.ts [hunk 2]
```diff
@@ -7,86 +7,82 @@ import * as vscode from 'vscode';
...
         let url = '';
         if (!context) {
             vscode.window.showErrorMessage("Cannot load the build config");
             return;
         }
         const consoleUrl = await Console.fetchOpenshiftConsoleUrl();
-        if (await Console.fetchClusterVersion() === null) {
-            url = `${consoleUrl}/k8s/ns/${context.id}/buildconfigs`;
+        if (consoleUrl['stdout']) {
+            url = `${Console.openShift4ClusterUrl(consoleUrl)}/k8s/ns/${context.name}/buildconfigs`;
```
Comment (dgolovin) [inline src/k8s/console.ts]: This does not work for 4.2.2. It should be
``` JavaScript
url = `${Console.openShift4ClusterUrl(consoleUrl)}/k8s/ns/${project}/buildconfigs/${context.name}`;
```

The same applies to other resources in OpenShift 4.2.2 cluster.

File: src/k8s/console.ts [hunk 3]
```diff
@@ -7,86 +7,82 @@ import * as vscode from 'vscode';
...
         let url = '';
         if (!context) {
             vscode.window.showErrorMessage("Cannot load the deployment config");
             return;
         }
         const consoleUrl = await Console.fetchOpenshiftConsoleUrl();
-        if (await Console.fetchClusterVersion() === null) {
-            url = `${consoleUrl}/k8s/ns/${context.id}/deploymentconfigs`;
+        if (consoleUrl['stdout']) {
+            url = `${Console.openShift4ClusterUrl(consoleUrl)}/k8s/ns/${context.name}/deploymentconfigs`;
```
Comment (dgolovin) [inline src/k8s/console.ts]: See comment above

File: src/k8s/console.ts [hunk 4]
```diff
@@ -7,86 +7,82 @@ import * as vscode from 'vscode';
...
         if (!context) {
             vscode.window.showErrorMessage("Cannot load the image stream");
             return;
         }
-        const consoleUrl = await this.fetchOpenshiftConsoleUrl();
-        if (await Console.fetchClusterVersion() === null) {
-            url = `${consoleUrl}/k8s/ns/${context.id}/imagestreams`;
+        const consoleUrl = await Console.fetchOpenshiftConsoleUrl();
+        if (consoleUrl['stdout']) {
+            url = `${Console.openShift4ClusterUrl(consoleUrl)}/k8s/ns/${context.name}/imagestreams`;
```
Comment (dgolovin) [inline src/k8s/console.ts]: See comment above

File: src/k8s/console.ts [hunk 5]
```diff
@@ -7,86 +7,82 @@ import * as vscode from 'vscode';
...
         let url = '';
         if (!context) {
             vscode.window.showErrorMessage("Cannot load the Project");
             return;
         }
         const consoleUrl = await Console.fetchOpenshiftConsoleUrl();
-        if (await Console.fetchClusterVersion() === null) {
-            url = `${consoleUrl}/overview/ns/${context.id}`;
+        if (consoleUrl['stdout']) {
+            url = `${Console.openShift4ClusterUrl(consoleUrl)}/overview/ns/${context.name}`;
```
Comment (dgolovin) [inline src/k8s/console.ts]: this works, because context == project == namespace.

**Pull request merged**